pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -354,17 +354,39 @@ func doBridgeAclsDelete(
 			aclArgs := types.AppNetworkACLArgs{IsMgmt: false, BridgeName: ulStatus.Bridge,
 				VifName: ulStatus.Vif, BridgeIP: ulStatus.BridgeIPAddr, AppIP: ulStatus.AllocatedIPAddr,
 				UpLinks: status.IfNameList}
-			ruleList, err := deleteACLConfiglet(aclArgs, ulStatus.ACLRules)
+			rules := getNetworkACLRules(ctx, appID, ulStatus.Name)
+			ruleList, err := deleteACLConfiglet(aclArgs, rules.ACLRules)
 			if err != nil {
 				log.Errorf("NetworkInstance DeleteACL failed: %s\n",
 					err)
 			}
-			ulStatus.ACLRules = ruleList
+			setNetworkACLRules(ctx, appID, ulStatus.Name, ruleList)
 		}
 	}
 	return
 }
 
+func getNetworkACLRules(ctx *zedrouterContext, appID uuid.UUID, intf string) types.ULNetworkACLs {
+	tmpMap := ctx.NLaclMap[appID]
+	if tmpMap == nil {
```

Comment:
I guess this particular if block should be there inside setNetworkACLRules function.

Reply:
ok